core/field_slider.js

```diff
@@ -180,10 +180,10 @@ Blockly.FieldSlider.prototype.addSlider_ = function(contentDiv) {
     this.readout_ = elements[1];
   }
   this.slider_ = new goog.ui.Slider();
-  this.slider_.setMoveToPointEnabled(false);
   this.slider_.setMinimum(this.min_);
   this.slider_.setMaximum(this.max_);
-  if (this.step_) this.slider_.setUnitIncrement(this.step_);
+  this.slider_.setMoveToPointEnabled(!this.step_);
```

Comment:
setMoveToPoint makes it so it snaps to the position, so this would make it so it snaps directly to where you click if you click on the bar (which feels a lot better than 'move towards where you clicked in 10 unit increments' to me). Can revert or make it optional if we're not sure on that.